compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -5189,6 +5191,262 @@ TR::Register *OMR::Power::TreeEvaluator::directCallEvaluator(TR::Node *node, TR:
    return resultReg;
    }
 
+static TR::Register *inlineSimpleAtomicUpdate(TR::Node *node, bool isAddOp, bool isLong, bool isGetThenUpdate, TR::CodeGenerator *cg)
+   {
+   TR::Node *valueAddrChild = node->getFirstChild();
+   TR::Node *deltaChild = NULL;
+
+   TR::Register *valueAddrReg = cg->evaluate(valueAddrChild);
+   TR::Register *deltaReg = NULL;
+   TR::Register *newValueReg = NULL;
+   TR::Register *oldValueReg = cg->allocateRegister();
+   TR::Register *cndReg = cg->allocateRegister(TR_CCR);
+   TR::Register *resultReg = NULL;
+
+   bool isDeltaConstant = false;
+   bool isDeltaImmediate = false;
+   bool isDeltaImmediateShifted = false;
+   bool reuseDeltaReg = false;
+
+   int32_t numDeps = 3;
+
+   int32_t delta = 0;
+
+   // Memory barrier --- NOTE: we should be able to do a test upfront to save this barrier,
+   //                          but Hursley advised to be conservative due to lack of specification.
+   generateInstruction(cg, TR::InstOpCode::lwsync, node);
+
+   TR::LabelSymbol *doneLabel = TR::LabelSymbol::create(cg->trHeapMemory(),cg);
+   TR::LabelSymbol *loopLabel = TR::LabelSymbol::create(cg->trHeapMemory(),cg);
+
+   loopLabel->setStartInternalControlFlow();
+   deltaChild = node->getSecondChild();
+
+   // Determine whether immediate instruction can be used.  For 64 bit operand,
+   // value has to fit in 32 bits to be considered for immediate operation
+   if (deltaChild->getOpCode().isLoadConst()
+          && !deltaChild->getRegister()
+          && (deltaChild->getDataType() == TR::Int32
+                 || (deltaChild->getDataType() == TR::Int64
+                        && deltaChild->getInt() == deltaChild->getLongInt())))
+      {
+      delta = (int32_t) deltaChild->getInt();
+      isDeltaConstant = true;
+
+      //determine if the constant can be represented as an immediate
+      if (delta <= UPPER_IMMED && delta >= LOWER_IMMED)
+         {
+         // avoid evaluating immediates for add operations
+         isDeltaImmediate = true;
+         }
+      else if (delta & 0xFFFF == 0)
+         {
+         // avoid evaluating shifted immediates for add operations
+         isDeltaImmediate = true;
+         isDeltaImmediateShifted = true;
+         }
+      else
+         {
+         numDeps++;
+         deltaReg = cg->evaluate(deltaChild);
+         }
+      }
+   else
+      {
+      numDeps++;
+      deltaReg = cg->evaluate(deltaChild);
+      }
+
+   generateLabelInstruction(cg, TR::InstOpCode::label, node, loopLabel);
+
+   if (isDeltaImmediate && !isAddOp)
+      {
+      // If argument is immediate, but the operation is not an add,
+      // the value must still be loaded into a register
+      // If argument is an immediate value and operation is an add,
+      // it will be used as an immediate operand in an add immediate instruction
+      numDeps++;
+      deltaReg = cg->allocateRegister();
+      loadConstant(cg, node, delta, deltaReg);
+      }
+
+   uint8_t len = isLong ? 8 : 4;
```

Comment:
This should be `const`

Reply:
Corrected in commit https://github.com/eclipse/omr/pull/3018/commits/cf67e3ee2bc9ec34b0170de4e4dc302555d71b9a